File: LoopConstruction/C/SquareRoot.c
#include <stdio.h>

int squareRoot(int x) {
    if (x == 0 || x == 1)
        return x;

    int start = 1, end = x, result = 0;
    while (start <= end) {
        int mid = start + (end - start) / 2;

        if (mid <= x / mid) {
            result = mid;
            start = mid + 1;
        } else {
            end = mid - 1;
        }
    }
    return result;
}

int main() {
    int x;
    printf("Enter a non-negative integer: ");
    scanf("%d", &x);

    if (x < 0) {
        printf("Invalid input! Please enter a non-negative integer.\n");
        return 0;
    }

    printf("Square root of %d is: %d\n", x, squareRoot(x));
    return 0;
}

File: LoopConstruction/Java/SquareRoot.java

import java.util.Scanner;

public class SquareRoot {
 public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    System.out.println("Enter a value : ");
    int x = sc.nextInt();


   if(x<0){
    System.out.println("Negative vlaue");
   }else{
    if (x == 0 || x == 1) {
        System.out.println(x);
        return;
    }

    int start = 1, end = x, result = 0;
    while (start <= end) {
        int mid = start + (end - start) / 2;

        if (mid <= x / mid) {
            result = mid;
            start = mid + 1;
        } else {
            end = mid - 1;
        }
    }
    System.out.println(result);
    return;

   }

 }   
}
